### PR TITLE
create connection health checks configuration

### PR DESCRIPTION
**Pull request by zoewangg**
## Description
create `ConnectionHealthChecksConfiguration` and expose it in the crt client builder

## Motivation and Context
https://github.com/aws/aws-sdk-java-v2/blob/master/docs/design/http-clients/aws-crt-client/DecisionLog.md#82420

## Testing
added unit tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [ ] I have read the **CONTRIBUTING** document
- [ ] Local run of `mvn install` succeeds
- [ ] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [ ] I have read the **README** document
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/AwsCrtAsyncHttpClient.java [hunk 1]
```diff
@@ -105,13 +106,25 @@ private AwsCrtAsyncHttpClient(DefaultBuilder builder, AttributeMap config) {
...
     }
 
+    private HttpMonitoringOptions revolveHttpMonitoringOptions(ConnectionHealthChecksConfiguration config) {
+        if (config == null) {
+            return null;
+        }
+
+        HttpMonitoringOptions httpMonitoringOptions = new HttpMonitoringOptions();
+        httpMonitoringOptions.setMinThroughputBytesPerSecond(config.minThroughputInBytesPerSecond());
+        int seconds = (int) (TimeUnit.MILLISECONDS.toSeconds(config.allowableThroughputFailureInterval().toMillis()));
```
Comment (dagnir) [inline http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/AwsCrtAsyncHttpClient.java:123]: `Duration` has `getSeconds()`

File: http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/ConnectionHealthChecksConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+    /**
+     * @return the minimum amount of throughput, in bytes per second, for a connection to be considered healthy.
+     */
+    public long minThroughputInBytesPerSecond() {
+        return minThroughputInBytesPerSecond;
+    }
+
+    /**
+     * @return How long, in seconds, a connection is allowed to be unhealthy before getting shut down.
```
Comment (dagnir) [inline http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/ConnectionHealthChecksConfiguration.java:46]: nit: `Duration` doesn't have a single unit so I don't think it would make sense to say "in seconds" here. We might want to add a note on the setter that it only supports seconds precision though

Comment (zoewangg) [reply http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/ConnectionHealthChecksConfiguration.java:46]: Yup, copy paste error

File: http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/AwsCrtAsyncHttpClient.java (hunk 1 above)
Comment (zoewangg) [reply http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/AwsCrtAsyncHttpClient.java:123]: `getSeconds` only get the seconds from the duration, i.e., not the total number of whole seconds in the duration

`toSeconds` is added since 9,, so we can't use it unfortunately. :(
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/time/Duration.html#toSeconds()

Comment (dagnir) [reply http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/AwsCrtAsyncHttpClient.java:123]: http://hg.openjdk.java.net/jdk9/jdk9/jdk/file/7f644a5d554a/src/java.base/share/classes/java/time/Duration.java they seem to have the same impl :P

Comment (dagnir) [reply http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/AwsCrtAsyncHttpClient.java:123]: don't feel strongly about this though

Comment (zoewangg) [reply http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/AwsCrtAsyncHttpClient.java:123]: Interesting! Didn't know they have the same impl. Why do they have two APIs then?! 🤔

Comment (zoewangg) [reply http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/AwsCrtAsyncHttpClient.java:123]: Found this relevant link: https://bugs.openjdk.java.net/browse/JDK-8142936 
TL;DR They added `toSeconds` to be consistent with other `toXXX` methods
